### DISCUSSION TITLE
Writing data from image data type column to file

### DISCUSSION THREAD

**Discussion by JoeFox82** [nextgenhealthcare/connect#6085] [Q&A]
Hello everyone,

I would like to use Mirth to write Microsoft Word files that are stored in a MS SQL Server image data type column as a file to the hard disc. 

To do this, I have queried the column in the Destination Transformer and want to save the data with the FileUtil:

`FileUtil.write("D:/test/test.doc", false, result.getBytes(1));
`

The files will be between 70 and 300kb in size, which is quite a correct size. However, the files are apparently all corrupt.

 It doesn't matter whether I cast the column with _CAST (imageColumn as VARBINARY(max))_ or not, the result is the same.

Any ideas?

Kind regards,
Jochen

**Comment by rjkroll** [comment]
I believe FileUtil just will write it all as a string of whatever it pulled from the column.   You'd have to do something like the following:
```
var imageColumn = result.getString('your_image_column_name'); // 

if (imageColumn != null) {
    var imageData = imageColumn.getBytes(1, imageColumn.length());
    var filePath = 'D:\\test\\test.doc';
    try {
        var fileOutputStream = new java.io.FileOutputStream(filePath);
        fileOutputStream.write(imageData);
        fileOutputStream.close();

        logger.info('Image successfully written to file: ' + filePath);
    } catch (e) {
        logger.error('Error writing image to file: ' + e.toString());
    }
} else {
    logger.warn('Image column is null');
}
```

**Reply by JoeFox82** [reply]
Hi rjkroll,

the result is the same as using the FileUtil. I'll post the way I've tested tomorrow

Kind regards,
Jochen

**Reply by pacmano1** [reply]
https://forums.mirthproject.io/forum/mirth-connect/support/183603-generate-file-from-db-blob seems another way to handle it.

**Reply by JoeFox82** [reply]
Hi pacmano1,

thank you very much, I've tried this way this morning!

All the ways to save the data in a file end with the same result. The number of bytes and the bytes themselves are always the same. I now think that the content of this column or the document itself are encrypted.

**Reply by Trilliakumaresan** [reply]
Hi pacmano1, 

Thank you for your sharing. It helps!
